src/main/kotlin/it/pagopa/wallet/services/PaymentMethodsService.kt

```diff
@@ -37,19 +48,28 @@ class PaymentMethodsService(
     private fun retrievePaymentMethodByApi(paymentMethodId: String): Mono<PaymentMethodResponse> =
         ecommercePaymentMethodsClient
             .getPaymentMethodById(paymentMethodId)
-            .flatMap { savePaymentMethodIntoCache(it) }
+            .doOnSuccess {
+                val emitResult = paymentMethodCacheSaveSink.tryEmitNext(it)
```

Comment:
The method tryEmitNext could throw an unchecked exception, should we take care of it here to avoid propagating it?

Reply:
Done here 766155debc365d9e91a13a05613e9702bce6de31

Reply:
Fix emit non fired here 8f2634e54a43f846f5f425ea2d59830a48b688df

Reply:
Is already there a test to check if the error handling works as expected?

Reply:
Done here ce1c34e0863aabe56c194c19bc57dfc632e6eab5